components/org.wso2.carbon.identity.authz.valve/src/main/java/org/wso2/carbon/identity/authz/valve/AuthorizationValve.java

```diff
@@ -120,6 +136,51 @@ public void invoke(Request request, Response response) throws IOException, Servl
         }
     }
 
+    private AuthorizationResult authorizeInOrganizationLevel(Request request, Response response,
+                                              AuthenticationContext authenticationContext,
+                                              ResourceConfig resourceConfig, AuthorizationContext authorizationContext)
+            throws IOException {
+
+        AuthorizationResult authorizationResult = new AuthorizationResult(AuthorizationStatus.DENY);
+
+        if (!isUserEmpty(authenticationContext)) {
+            String httpMethod = request.getMethod();
+            String tenantDomainFromURLMapping = Utils.getTenantDomainFromURLMapping(request);
+
+            OrganizationManagementAuthorizationContext orgMgtAuthorizationContext =
+                    new OrganizationManagementAuthorizationContext();
+
+            if (resourceConfig != null) {
+                if(StringUtils.isNotEmpty(resourceConfig.getPermissions())) {
+                    orgMgtAuthorizationContext.setPermissionString(resourceConfig.getPermissions());
+                }
+                if (CollectionUtils.isNotEmpty(resourceConfig.getScopes())) {
+                    orgMgtAuthorizationContext.setRequiredScopes(resourceConfig.getScopes());
+                }
+                orgMgtAuthorizationContext.setContext(resourceConfig.getContext());
+            }
+            orgMgtAuthorizationContext.setHttpMethods(httpMethod);
+            orgMgtAuthorizationContext.setUser(authenticationContext.getUser());
+            orgMgtAuthorizationContext.setTenantDomainFromURLMapping(tenantDomainFromURLMapping);
+            orgMgtAuthorizationContext.addParameter(OAUTH2_ALLOWED_SCOPES,
+                    authenticationContext.getParameter(OAUTH2_ALLOWED_SCOPES));
+            orgMgtAuthorizationContext.addParameter(OAUTH2_VALIDATE_SCOPE,
+                    authenticationContext.getParameter(OAUTH2_VALIDATE_SCOPE));
+
+            List<AuthorizationManager> authorizationManagerList = AuthorizationValveServiceHolder.getInstance()
+                    .getAuthorizationManagerList();
+            AuthorizationManager authorizationManager = HandlerManager.getInstance()
+                    .getFirstPriorityHandler(authorizationManagerList, true);
+            try {
+                authorizationResult = authorizationManager.authorize(orgMgtAuthorizationContext);
+            } catch (AuthzServiceServerException e) {
+                APIErrorResponseHandler.handleErrorResponse(authenticationContext, response,
+                        HttpServletResponse.SC_BAD_REQUEST, null);
```

Comment:
is `HttpServletResponse.SC_BAD_REQUEST` the correct status code here?

Reply:
fixed in [aea691b](https://github.com/wso2-extensions/identity-carbon-auth-rest/pull/184/commits/aea691baa6bd686a7f906d6ae3785f847309c754)